### PR TITLE
Update eslint and prettier

### PR DESCRIPTION
**Pull request by vsumner**
## Description
This PR updates `eslint` and `prettier` to latest and changes `eslint-plugin-shopify` to `@shopify/eslint-plugin`.

## Type of change

- [ ]  Patch: Bug/ Documentation fix (non-breaking change which fixes an issue or adds documentation)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -882,13 +882,20 @@
...
 
-"@babel/runtime@^7.1.2", "@babel/runtime@^7.4.0", "@babel/runtime@^7.4.5":
+"@babel/runtime@^7.1.2", "@babel/runtime@^7.4.0":
   version "7.6.3"
   resolved "https://registry.yarnpkg.com/@babel/runtime/-/runtime-7.6.3.tgz#935122c74c73d2240cafd32ddb5fc2a6cd35cf1f"
   integrity sha512-kq6anf9JGjW8Nt5rYfEuGRaEAaH1mkv3Bbu6rYvLOpPh/RusSJXuKPEAoZ7L7gybZkchE8+NV5g9vKF4AGAtsA==
   dependencies:
     regenerator-runtime "^0.13.2"
 
+"@babel/runtime@^7.4.5":
```
Comment (jgodson) [inline yarn.lock:892]: can we dedupe this?

File: yarn.lock [hunk 2]
```diff
@@ -4610,27 +4714,35 @@ escodegen@^1.11.1, escodegen@^1.9.1:
...
-  integrity sha512-H6DOj+ejw7Tesdgbfs4jeS4YMFrT8uI8xwd1gtQqXssaR0EQ26L+2O/w6wkYFy2MymON0fTwHmXBvvfLNZVZEw==
+eslint-module-utils@2.5.0:
+  version "2.5.0"
+  resolved "https://registry.yarnpkg.com/eslint-module-utils/-/eslint-module-utils-2.5.0.tgz#cdf0b40d623032274ccd2abd7e64c4e524d6e19c"
+  integrity sha512-kCo8pZaNz2dsAW7nCUjuVoI11EBXXpIzfNxmaoLhXoRDOnqXLC4iSGVRdZPhOitfbdEfMEfKOiENaK6wDPZEGw==
+  dependencies:
+    debug "^2.6.9"
+    pkg-dir "^2.0.0"
+
+eslint-module-utils@^2.4.1:
```
Comment (jgodson) [inline yarn.lock:4740]: worth deduping? Not sure how much it matters here but 2.5 would meet this too

File: yarn.lock [hunk 3]
```diff
@@ -10005,13 +10226,20 @@ resolve@1.1.7:
...
 
-resolve@1.x, resolve@^1.1.6, resolve@^1.1.7, resolve@^1.10.0, resolve@^1.10.1, resolve@^1.11.0, resolve@^1.12.0, resolve@^1.15.1, resolve@^1.3.2, resolve@^1.5.0:
+resolve@1.x, resolve@^1.1.6, resolve@^1.1.7, resolve@^1.10.0, resolve@^1.11.0, resolve@^1.15.1, resolve@^1.3.2:
   version "1.15.1"
   resolved "https://registry.yarnpkg.com/resolve/-/resolve-1.15.1.tgz#27bdcdeffeaf2d6244b95bb0f9f4b4653451f3e8"
   integrity sha512-84oo6ZTtoTUpjgNEr5SJyzQhzL72gaRodsSfyxC/AXRvwu0Yse9H8eF9IpGo7b8YetZhlI6v7ZQ6bKBFV/6S7w==
   dependencies:
     path-parse "^1.0.6"
 
+resolve@^1.10.1, resolve@^1.12.0, resolve@^1.13.1:
```
Comment (jgodson) [inline yarn.lock:10236]: looks like this could get deduped too

File: yarn.lock [hunk 4]
```diff
@@ -11160,11 +11456,16 @@ ts-node@^8:
     source-map-support "^0.5.6"
     yn "^3.0.0"
 
-tslib@^1, tslib@^1.10.0, tslib@^1.7.1, tslib@^1.8.1, tslib@^1.9.0, tslib@^1.9.3:
+tslib@^1, tslib@^1.10.0, tslib@^1.7.1, tslib@^1.9.0, tslib@^1.9.3:
```
Comment (jgodson) [inline yarn.lock:11459]: dedupe here too

File: yarn.lock (hunk 2 above)
Comment (vsumner) [reply yarn.lock:4708]: This is actually `2.6.0` the `2.5.0` above is locked because of `@shopify/eslint-plugin` It will need an update.

**Pull request merged**